lib/stdlib/src/sets.erl

```diff
@@ -74,56 +84,74 @@
 
 -type set() :: set(_).
 
--opaque set(Element) :: #set{segs :: segs(Element)}.
+-opaque set(Element) :: #set{segs :: segs(Element)} | #{Element => ?VALUE}.
 
 %%------------------------------------------------------------------------------
 
 %% new() -> Set
 -spec new() -> set().
-new() ->
+new() -> #{}.
+
+%% from_list([Elem]) -> Set.
+%%  Build a set from the elements in List.
+-spec from_list(List) -> Set when
+      List :: [Element],
+      Set :: set(Element).
+from_list(Ls) -> maps:from_list([{K,?VALUE}||K<-Ls]).
+
+%%------------------------------------------------------------------------------
+
+deprecated_new() ->
     Empty = mk_seg(?seg_size),
     #set{empty = Empty, segs = {Empty}}.
 
+deprecated_from_list(L) ->
+    lists:foldl(fun (E, S) -> add_element(E, S) end, deprecated_new(), L).
+
+%%------------------------------------------------------------------------------
+
 %% is_set(Set) -> boolean().
 %%  Return 'true' if Set is a set of elements, else 'false'.
 -spec is_set(Set) -> boolean() when
       Set :: term().
+is_set(#{}) -> true;
```

Comment:
Would it make sense wrapping the map into a record, say `map_set`, so as to avoid any map being interpreted as a set?